src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
+    let i = 0;
+
+    if (arguments.length < 2) {
+      prev = this[0];
+      i = 1;
+    }
+
+    for (i; i < this.length; i++) {
+      prev = callback(prev, this[i], i, this);
+    }
```

Comment:
We on `Array iteration methods` topic. So maybe should use some array method instead of a simple loop )